src/cromshell/utilities/CromshellConfig.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import sys
+from pathlib import Path
+
+
+class CromshellConfig:
+    """Setup Cromshell config details. Intended to be used as a singleton"""
+
+    @classmethod
+    def user_defined_slim_metadata_parameters(cls, slim_metadata_parameters_user):
+        """Override Cromwell Slim Metadata Key From Command Line"""
+        if slim_metadata_parameters_user:
+            cls.slim_metadata_parameters = (
+                slim_metadata_parameters_user
+                + "&includeKey=subWorkflowMetadata&includeKey=subWorkflowId"
+            )
+
+    @classmethod
+    def override_cromwell_config_server(cls, server_user=None, workflow_id=None):
+        """
+        Override Cromwell Server From Command Line or Environment or Submission file
+
+        This function allows users to override the cromwell server set in the
+        cromshell config file using the command line options or environment
+        variables. By default the server URL in the config file is used, however
+        if an URL is specified in the command line or environment then this
+        overrides the default. If a workflow Id is provided without a server URL
+        then the server associated with a workflow id in the submission file
+        overrides the default. If a workflow ID is provided but isn't found
+        in the submission file then the default server is used.
+        """
+
+        if server_user is None and workflow_id is None:
+            cls.LOGGER.info(
+                "Workflow id and cromwell server not specified. Using default"
+            )
+        elif server_user:
+            cls.cromwell_server = server_user
+            CromshellConfig.LOGGER.info(
+                "Cromwell server set to the provided in cli or environment argument."
```

Comment:
This sentence is weird.  Maybe "Cromwell server url was overridden by command line argument or environment variable". (it would be good to specify which)

Reply:
we're moving away from the environment table so I'll just say the command line.
done